services/service-tester.js

```diff
@@ -29,6 +31,32 @@ class ServiceTester {
     })
   }
 
+  static forServiceClass(ServiceClass) {
+    const id = ServiceClass.name
+    const pathPrefix = `/${ServiceClass.url.base}`
+    return new this({
+      id,
+      title: id,
+      pathPrefix,
+    })
+  }
+
+  static forThisService() {
```

Comment:
It seems what we're trying to do here is create a `ServiceTester` object based on some non-trivial criteria. I think a more common approach here would be to use the factory pattern and have something like a `ServiceTesterFactory` which is responsible for producing the `ServiceTester` object. Having a static method on the `ServiceTester` class which creates instances of itself seems like a more unusual structure.

What do you think?

Reply:
It's a subtle point you're making, though I agree, that given how specific this logic is it's reasonable to move this out of the class. I don't think we need a factory _class_, though we could move the magical factory function into its own module and do something like:

```js
const createServiceTester = require('../create-service-tester')
```

or 

```js
const { automagicallyCreateServiceTester } = require('../create-service-tester')
```